docs/general/how-ra-works.md

```diff
@@ -0,0 +1,45 @@
+# How RetroAchievements Works
+
+RetroAchievements (RA) provides users the ability to earn achievements in retro games from an RA achievement set.  It awards achievements by comparing a game’s memory, henceforth referred to as RAM, to achievement code written by an RA developer.  Achievement code, also known as logic, is a list of memory conditions chosen by the developer, that when all are simultaneously true on a single frame, will award an achievement.
+
+## What Is an Achievement Set?
+
+An achievement set is the compilation of individual achievements, Rich Presence and leaderboards for a particular game.  Achievements are comprised of achievement logic, a title, a description, a point value and a badge.  Additionally, a set must contain a Rich Presence script which provides site users information on where active players are in a game.  Sets may have leaderboards that track how well players perform in certain things in a game such as how quickly they can beat stages, how many points they can score and many other things.  RA users interface with all these components and subcomponents.  Each one is a vital part of the player experience.
+
+## What Is Achievement Code and How Is It Written?
+
+RA code is a custom text-based language that is interpreted by RA supporting emulators. This code is constructed by developers with one or two achievement development tools.  Developers may use whichever tool they prefer. The Achievement Editor in the RAIntegration toolkit requires no prior knowledge of coding. On the other hand, RATools is a standalone executable designed for developers who find it more comfortable to write in a scripting language to develop their achievement set. Both the Achievement Editor and RATools produce similar code that is used for all achievements, leaderboards and Rich Presence.
```

Comment:
Wherever possible, the first mention of a concept should link to its relevant docs page if we have one. For example, the first mention of "RATools" should link to its corresponding docs page.

This has numerous benefits:
* It helps the reader jump to a concept they may want to learn more about.
* It trains the built-in search feature that certain concepts may be connected.
* It trains external search engines like Google that certain concepts are connected, resulting in better external search results.

Only linking it the first time is an important nuance. Repeated linking to the same place on the same page actually has the opposite effect - the algorithms might assume either a mistake has been made, or they assume the associated connection is so common that it's "understood" by readers and not actually special.

Reply:
Much thanks for the feedback and education!  Added RATools doc link as recommended.